src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentSelector.cs

```diff
@@ -3,44 +3,41 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
-using System.Reflection;
 using Microsoft.AspNet.Mvc.Core;
 using Microsoft.Framework.Internal;
 
-namespace Microsoft.AspNet.Mvc
+namespace Microsoft.AspNet.Mvc.ViewComponents
 {
+    /// <summary>
+    /// Default implementation of <see cref="IViewComponentSelector"/>.
+    /// </summary>
     public class DefaultViewComponentSelector : IViewComponentSelector
     {
-        private readonly IAssemblyProvider _assemblyProvider;
+        private readonly IViewComponentDescriptorCollectionProvider _descriptorProvider;
 
-        private ViewComponentCandidateCache _cache;
+        private ViewComponentDescriptorCache _cache;
 
-        public DefaultViewComponentSelector(IAssemblyProvider assemblyProvider)
+        /// <summary>
+        /// Creates a new <see cref="DefaultViewComponentSelector"/>.
+        /// </summary>
+        /// <param name="descriptorProvider">The <see cref="IViewComponentDescriptorCollectionProvider"/>.</param>
+        public DefaultViewComponentSelector(IViewComponentDescriptorCollectionProvider descriptorProvider)
         {
-            _assemblyProvider = assemblyProvider;
+            _descriptorProvider = descriptorProvider;
         }
 
-        public Type SelectComponent([NotNull] string componentName)
+        /// <inheritdoc />
+        public ViewComponentDescriptor SelectComponent([NotNull] string componentName)
         {
-            if (_cache == null)
+            var collection = _descriptorProvider.ViewComponents;
+            if (_cache == null || _cache.Version != collection.Version)
```

Comment:
Still don't understand the version stuff. In the context of using the default MVC ViewComponent bits `.Version` will always be `0`.